csharp-basics/exercises/Scooters/Scooters/RentalCompany.cs

```diff
@@ -0,0 +1,108 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections;
+
+namespace Scooters
+{
+    public partial class RentalCompany : IRentalCompany
+    {
+        private string _name;
+        public IScooterService _scooterService = new ScooterService();
+        List<RentalData> _rentalHistory = new List<RentalData>();        
+
+        public RentalCompany(string CompanyName)
+        {
+            _name = CompanyName;
+        }
+
+        public RentalCompany(string CompanyName, ScooterService scooterService, List<RentalData> rentalHistory)
+        {
+            _name = CompanyName;
+            _scooterService = scooterService;
+            _rentalHistory = rentalHistory;
+        }
+
+        public string Name()
+        {
+           return _name;          
+        }
+
+        public void StartRent(string id)
+        {
+            Scooter scooter = _scooterService.GetScooterById(id);
+            if (scooter.IsActive)
+            {
+                if (!scooter.IsRented)
+                {
+                    scooter.RentalStartTime = DateTime.Now;
+                    scooter.IsRented = true;
+                }
+                else throw new ScooterRentedException("Scooter is already under rent!");
+            }
+            else throw new ScooterActivityException("Scooter is not active!");
+        }
+
+        public decimal EndRent(string id)
+        {
+            decimal rentPrice = 0m;
+            Scooter scooter = _scooterService.GetScooterById(id);
+            if (scooter.IsActive)
```

Comment:
Scooter klasē nebija manuprāt IsActive property.

Reply:
Pievienoju šo, lai nebūtu pilnīgi jādzēš Scooter no scooterList. Tas, mamuprāt, nav optimāli pavisam izdzēst skuteri.  Galvenokārt, scooterserviss izsekojamībai, un lai tam pašam Id, kas tika izdzēsts netiktu peivienots jauns scooter ar tādu pašu id, ieviešot tādā veidā jucekli RentalCompany RentalHistory ar skuteriem ar vienādiem id.

Reply:
Protams, es varu izdarīt vnk ar List.Remove(). , bet nav sagaidāms, ka skuteri tiks dzēsti nesaskaitāmos apjomos (ka šis lists pārplūdīs), līdz ar to, manuprāt, pareizāk ir saglabāt datus par izņemamo skuteri, papildus saglābājot iespēju salauzto vai nozagto skuteri aktivizēt atkal, ja tas tiek salabots, vai atrasts.
Ja tā neuzskati, nomainīšu.